### PR TITLE
First commits

### LINKED ISSUE #4: Replace Celery Flower Image
# 💡 Summary #

The current image -- `crgwbr/docker-celery-flower` --  is outdated. This most recent image is 4 years old, and the GitHub repository that built it is dormant, so further updates are unlikely.

## Motivation and context ##

First noted in cisagov/admiral#17. We should be using the latest stable image and version wherever possible.

### LINKED ISSUE #40: Separate Docker components
# 💡 Summary #

We need to create an `admiral-docker` repository. 

## Motivation and context ##

We want to decouple the Python and Docker components to this project. This repository descends from [`skeleton-python`](https://github.com/cisagov/skeleton-python), so it makes sense to move the Docker bits into a Docker repository.

## Implementation notes ##

The new repository should descend from [`skeleton-docker`](https://github.com/cisagov/skeleton-docker).

## Acceptance criteria ##

- [x] There exists an `admiral-docker` repository that descends from [`skeleton-docker`](https://github.com/cisagov/skeleton-docker).
- [ ] The Docker components found in this repository are present in `admiral-docker`.

### PR DESCRIPTION
**Pull request by king-alexander**
#  #

## 🗣 Description ##
These are the first commits towards a repository decomposition for the [Admiral](https://github.com/cisagov/admiral). They generally pull in Admiral's Docker components and change defaults inherited from the [skeleton repository](https://github.com/skeleton-docker).

## 💭 Motivation and context ##
It's desirable to separate the containerization logic from a project's content. This pull request is a step towards decomposing [Admiral](https://github.com/cisagov/admiral) into smaller, more manageable repositories. 

- Closes #4: I used the opportunity to pick off some low-hanging fruit. The [`mher/flower` image](https://hub.docker.com/r/mher/flower/) was last updated 3 months ago.
- Resolves cisagov/admiral#40

## 🧪 Testing ##

I modified the standard tests inherited from [cisagov/skeleton-docker](https://github.com/cisagov/skeleton-docker) to cover my changes to the Docker configuration. Not all tests were easy to modify. In particular, the `main_container` tests do not transfer well, as there is no main container in the Docker composition. Implementing these tests is scoped as future work in #6. 

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [x] Revert dependencies to default branches.
- [x] Finalize version.

## ✅ Post-merge checklist ##

- [x] Add a tag or create a release.

### CODE REVIEW

**Comment by dav3r** [comment]
FYI, the linting failures ([here](https://github.com/cisagov/admiral-docker/actions/runs/4255133166/jobs/7402348369) and [here](https://github.com/cisagov/admiral-docker/actions/runs/4255132899/jobs/7402347799)) should be resolved after https://github.com/cisagov/admiral-docker/pull/7 has been approved and merged into this branch.

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
-ARG VERSION=unspecified
+ARG VERSION=1.3.0
```
Comment (dav3r) [inline Dockerfile]: The `VERSION` argument should be left unspecified- it is meant to be passed in when the image is created, as noted in the [README](https://github.com/cisagov/guacscanner-docker/blob/develop/Dockerfile#L1).  You can also see how the version is [determined](https://github.com/cisagov/admiral-docker/blob/develop/.github/workflows/build.yml#L182-L219) and [used](https://github.com/cisagov/admiral-docker/blob/develop/.github/workflows/build.yml#L253-L254) in the build workflow.
```suggestion
ARG VERSION=unspecified
```

**Review by mcdonnnj — DISMISSED** [review]
Before you make any other changes please update the image name:
https://github.com/cisagov/admiral-docker/blob/93969b2d79be588163a78b1ffd460ec986293ce5/.github/workflows/build.yml#L30

**Review by dav3r — CHANGES_REQUESTED** [review]

**Comment by dav3r** [comment]
@king-alexander Now that #7 has been merged, you should pull those changes in to this branch and deconflict them.  Before you do that though, make sure you do what @mcdonnnj said in [his comment](https://github.com/cisagov/admiral-docker/pull/5#pullrequestreview-1314081139).  That should get you closer to passing all of the checks in this PR.

File: Dockerfile [hunk 2]
```diff
@@ -26,7 +21,10 @@ ARG CISA_UID=421
 ARG CISA_GID=${CISA_UID}
 ARG CISA_USER="cisa"
 ENV CISA_GROUP=${CISA_USER}
-ENV CISA_HOME="/home/${CISA_USER}"
+ENV CISA_HOME="/home/${CISA_USER}" \
+    ADMIRAL_CONFIG_FILE="/run/secrets/admiral.yml" \
+    ADMIRAL_CONFIG_SECTION="dev-mode" \
+    ADMIRAL_WORKER_NAME="dev"
```
Comment (dav3r) [inline Dockerfile]: Admiral-related environment variables should be moved to their own section and specified one per line:
```suggestion
ENV CISA_HOME="/home/${CISA_USER}"

###
# Admiral configuration variables
###
ENV ADMIRAL_CONFIG_FILE="/run/secrets/admiral.yml"
ENV ADMIRAL_CONFIG_SECTION="dev-mode"
ENV ADMIRAL_WORKER_NAME="dev"
```

File: README.md [hunk 3]
```diff
@@ -6,189 +6,95 @@
...
 
 ## Image tags ##
 
 The images of this container are tagged with [semantic
-versions](https://semver.org) of the underlying example project that they
-containerize.  It is recommended that most users use a version tag (e.g.
-`:0.0.1`).
+versions](https://semver.org) of the [admiral](https://github.com/cisagov/admiral)
+Python library that they containerize.  It is recommended that most users
+use a version tag (e.g. `:0.0.1`).
```
Comment (dav3r) [inline README.md]: Versions in the README should match the value in `src/version.txt`:
```suggestion
use a version tag (e.g. `:1.3.0`).
```

File: README.md [hunk 4]
```diff
@@ -6,189 +6,95 @@
...
-There are no required environment variables.
-
-<!--
 | Name  | Purpose | Default |
 |-------|---------|---------|
-| `REQUIRED_VARIABLE` | Describe its purpose. | `null` |
--->
+| `ADMIRAL_CONFIG_FILE` | Celery configuration | `admiral.yml` |
+| `CISA_HOME` | Home folder | `/home/cisa` |
+| `CISA_GROUP` | Group identifier | `cisa` |
```
Comment (dav3r) [inline README.md]: Any variable that has a default value is optional, by definition.

File: README.md [hunk 5]
```diff
@@ -6,189 +6,95 @@
...
-
-### Docker ###
-
-1. Stop the running container:
-
-    ```console
-    docker stop <container_id>
-    ```
-
-1. Pull the new image:
```
Comment (dav3r) [inline README.md:125]: Why did you remove the "Using secrets with your container" and "Updating your container" sections?  Those both seem like they would be useful to include in the README.

File: README.md [hunk 6]
```diff
@@ -197,8 +103,8 @@ Build the image locally using this git repository as the [build context](https:/
 ```console
 docker build \
   --build-arg VERSION=0.0.1 \
-  --tag cisagov/example:0.0.1 \
-  https://github.com/cisagov/example.git#develop
+  --tag cisagov/admiral-docker:0.0.1 \
```
Comment (dav3r) [inline README.md]: Use correct image name.  Also, versions in the README should match the value in src/version.txt:
```suggestion
  --tag cisagov/admiral:1.3.0 \
```

File: README.md [hunk 7]
```diff
@@ -6,189 +6,95 @@
...
-
-See the [tags tab](https://hub.docker.com/r/cisagov/example/tags) on Docker
+|`cisagov/admiral-docker:1.2.3`| An exact release version. |
+|`cisagov/admiral-docker:1.2`| The most recent release matching the major and minor version numbers. |
+|`cisagov/admiral-docker:1`| The most recent release matching the major version number. |
+|`cisagov/admiral-docker:edge` | The most recent image built from a merge into the `develop` branch of this repository. |
+|`cisagov/admiral-docker:nightly` | A nightly build of the `develop` branch of this repository. |
+|`cisagov/admiral-docker:latest`| The most recent release image pushed to a container registry.  Pulling an image using the `:latest` tag [should be avoided.](https://vsupalov.com/docker-latest-tag/) |
+
+See the [tags tab](https://hub.docker.com/r/cisagov/admiral-docker/tags) on Docker
```
Comment (dav3r) [inline README.md]: Use correct image name.  Also, versions in the README should match the value in src/version.txt:
```suggestion
|`cisagov/admiral:1.3.0`| An exact release version. |
|`cisagov/admiral:1.3`| The most recent release matching the major and minor version numbers. |
|`cisagov/admiral:1`| The most recent release matching the major version number. |
|`cisagov/admiral:edge` | The most recent image built from a merge into the `develop` branch of this repository. |
|`cisagov/admiral:nightly` | A nightly build of the `develop` branch of this repository. |
|`cisagov/admiral:latest`| The most recent release image pushed to a container registry.  Pulling an image using the `:latest` tag [should be avoided.](https://vsupalov.com/docker-latest-tag/) |

See the [tags tab](https://hub.docker.com/r/cisagov/admiral/tags) on Docker
```

File: README.md [hunk 8]
```diff
@@ -229,16 +135,9 @@ Docker:
       --platform linux/amd64 \
       --build-arg VERSION=0.0.1 \
       --output type=docker \
-      --tag cisagov/example:0.0.1 .
+      --tag cisagov/admiral-docker:0.0.1 .
```
Comment (dav3r) [inline README.md]: Use correct image name.  Also, versions in the README should match the value in src/version.txt:
```suggestion
      --tag cisagov/admiral:1.3.0 .
```

**Comment by dav3r** [comment]
> Before you make any other changes please update the image name:
> 
> https://github.com/cisagov/admiral-docker/blob/93969b2d79be588163a78b1ffd460ec986293ce5/.github/workflows/build.yml#L30

BTW, per our convention, the image should be called `admiral`, not `admiral-docker`.

File: README.md [hunk 9]
```diff
@@ -6,189 +6,95 @@
 
 ## Docker Image ##
 
-[![Docker Pulls](https://img.shields.io/docker/pulls/cisagov/example)](https://hub.docker.com/r/cisagov/example)
-[![Docker Image Size (latest by date)](https://img.shields.io/docker/image-size/cisagov/example)](https://hub.docker.com/r/cisagov/example)
+[![Docker Pulls](https://img.shields.io/docker/pulls/cisagov/admiral-docker)](https://hub.docker.com/r/cisagov/admiral-docker)
+[![Docker Image Size (latest by date)](https://img.shields.io/docker/image-size/cisagov/admiral-docker)](https://hub.docker.com/r/cisagov/admiral-docker)
```
Comment (dav3r) [inline README.md]: Use correct image name:
```suggestion
[![Docker Pulls](https://img.shields.io/docker/pulls/cisagov/admiral)](https://hub.docker.com/r/cisagov/admiral)
[![Docker Image Size (latest by date)](https://img.shields.io/docker/image-size/cisagov/admiral)](https://hub.docker.com/r/cisagov/admiral)
```

**Review by dav3r — CHANGES_REQUESTED** [review]
Noting a few more items:

File: README.md (hunk 4 above)
Comment (king-alexander) [reply README.md]: I edited [51c8336](https://github.com/cisagov/admiral-docker/pull/5/commits/51c83362baf155eb8f446af7ceb224215ac1559d) to put all variables under the _Optional_ heading.

File: README.md (hunk 5 above)
Comment (king-alexander) [reply README.md:125]: Good point! I was looking more closely at the Admiral [README](https://github.com/cisagov/admiral/README.md) when I put this one together. But I agree. I dropped the commit where I removed these sections.

File: README.md (hunk 4 above)
Comment (dav3r) [reply README.md]: Looks like this was taken care of in https://github.com/cisagov/admiral-docker/pull/5/commits/9f8b4140998172360a3e7593ea6a70827217ebd1. 👍

File: README.md (hunk 5 above)
Comment (dav3r) [reply README.md:125]: Thanks for doing that.  Next, you should customize the "Using secrets with your container" section to make it applicable and useful for this repo.

File: README.md [hunk 10]
```diff
@@ -114,81 +91,63 @@ environment variables.  See the
...
-### Required ###
-
-There are no required environment variables.
-
-<!--
-| Name  | Purpose | Default |
-|-------|---------|---------|
-| `REQUIRED_VARIABLE` | Describe its purpose. | `null` |
--->
-
```
Comment (dav3r) [inline README.md]: I would leave this section in.  If required environment variables are ever added in the future, this will make life easy for whoever handles that.

File: README.md [hunk 11]
```diff
@@ -114,81 +91,63 @@ environment variables.  See the
...
 ## Secrets ##
 
 | Filename     | Purpose |
 |--------------|---------|
-| `quote.txt` | Replaces the secret stored in the example library's package data. |
+| `admiral.yml` | Celery configuration |
+| `mongo.yml` | MongoDB configuration |
+| `mongo-root-passwd.txt` | MongoDB root password |
+| `redis.conf` | Redis configuration |
+| `sslmate-api-key.txt` | CT Search API key |
```
Comment (dav3r) [inline README.md]: It never hurts to be a bit more verbose in documentation:
```suggestion
| `sslmate-api-key.txt` | API key for SSLMate's Certificate Transparency Search API |
```

File: docker-compose.yml [hunk 12]
```diff
@@ -4,34 +4,104 @@ version: "3.7"
...
+    ADMIRAL_CONFIG_SECTION: dev-mode
+    ADMIRAL_WORKER_NAME: dev
+  image: cisagov/admiral
+  init: true
+  secrets:
+    - source: admiral_yml
+      target: admiral.yml
+    - source: sslmate_api_key_txt
+      target: sslmate-api-key.txt
+  # map for development only, comment out otherwise
```
Comment (dav3r) [inline docker-compose.yml]: Can this comment be clarified a bit more?  It's unclear to me what it refers to and when exactly someone would uncomment it.

File: docker-compose.yml [hunk 13]
```diff
@@ -4,34 +4,104 @@ version: "3.7"
...
+# volumes:
+  # - ./src/admiral:/usr/src/admiral/admiral
```
Comment (dav3r) [inline docker-compose.yml]: I believe the indentation is off here, which makes this section read oddly (even though it is commented out):
```suggestion
  # volumes:
    # - ./src/admiral:/usr/src/admiral/admiral
```

File: docker-compose.yml (hunk 12 above)
Comment (king-alexander) [reply docker-compose.yml]: Would it make more sense to remove this block? I suppose @felddy intended this for development of the Python source code, so we shouldn’t need it in this repository.

File: README.md (hunk 10 above)
Comment (king-alexander) [reply README.md]: I reintroduced the section in [174d98b](https://github.com/cisagov/admiral-docker/pull/5/commits/174d98b96e1d5378017059ce32360fc871f7be88).

File: README.md (hunk 5 above)
Comment (king-alexander) [reply README.md:125]: I edited [174d98b](https://github.com/cisagov/admiral-docker/pull/5/commits/174d98b96e1d5378017059ce32360fc871f7be88) to be more specific about how secrets are used in this repository. I thought the steps for creating a secret and adding it to the composition were irrelevant, but I can add them back in if you think they're useful.

File: docker-compose.yml (hunk 12 above)
Comment (dav3r) [reply docker-compose.yml]: If it's no longer useful, then I completely agree that it should be removed.

File: README.md (hunk 5 above)
Comment (dav3r) [reply README.md:125]: What you have in there now works for me - thanks!

File: docker-compose.yml (hunk 12 above)
Comment (king-alexander) [reply docker-compose.yml]: Removed in [dd8f2a6](https://github.com/cisagov/admiral-docker/pull/5/commits/dd8f2a627bfc38b5fd09598b27310489d5b28ee7).

**Comment by king-alexander** [comment]
I dropped support for all platforms except `linux/amd64` and `linux/arm64` in [58ba017](https://github.com/cisagov/admiral-docker/pull/5/commits/58ba017966def04449ffc1455b46dd03610b4aed). They should be reenabled after https://github.com/cisagov/skeleton-docker is updated to address the rust compiler issue.

File: .github/workflows/build.yml [hunk 14]
```diff
@@ -27,10 +27,9 @@ on:
 env:
   BUILDX_CACHE_DIR: ~/.cache/buildx
   CURL_CACHE_DIR: ~/.cache/curl
-  IMAGE_NAME: cisagov/example
+  IMAGE_NAME: cisagov/admiral
   PIP_CACHE_DIR: ~/.cache/pip
-  PLATFORMS: "linux/amd64,linux/arm/v6,linux/arm/v7,\
-  linux/arm64,linux/ppc64le,linux/s390x"
+  PLATFORMS: "linux/amd64, linux/arm64"
```
Comment (dav3r) [inline .github/workflows/build.yml:36]: You should include a comment here about why the platforms were dropped and when they can be re-added.  You should also create an issue to re-add them and reference that issue in your comment.

Comment (king-alexander) [reply .github/workflows/build.yml:36]: I created #8 and reference the issue in [6dc81cf](https://github.com/cisagov/admiral-docker/pull/5/commits/6dc81cf4474fc2156747c0e9723cd157c198e661).

Comment (dav3r) [reply .github/workflows/build.yml:36]: The comment looks good, however the commit that you reference in the #8 summary (https://github.com/cisagov/admiral-docker/pull/5/commits/58ba017966def04449ffc1455b46dd03610b4aed) is no longer correct.  You should edit it to mention https://github.com/cisagov/admiral-docker/pull/5/commits/6dc81cf4474fc2156747c0e9723cd157c198e661 instead.

Comment (king-alexander) [reply .github/workflows/build.yml:36]: Aha! Thanks for catching that. I updated the comment.

**Review by dav3r — APPROVED** [review]
I think this is looking real solid now - nice work! 🤠

File: Dockerfile [hunk 15]
```diff
@@ -1,6 +1,9 @@
-ARG VERSION=unspecified
+ARG PYTHON_IMAGE_VERSION=3.10.4-alpine
```
Comment (jsf9k) [inline Dockerfile]: This is a pretty old version of Python at this point.  The [most recent release is 3.11.2](https://hub.docker.com/_/python).  Is there any reason that we can't use that?

**Review by jsf9k — COMMENTED** [review]
I have one minor question.

File: Dockerfile (hunk 15 above)
Comment (king-alexander) [reply Dockerfile]: Not to my knowledge. Let me try with that version and see what happens.

Comment (king-alexander) [reply Dockerfile]: Updated the version in [d81a6d8](https://github.com/cisagov/admiral-docker/pull/5/commits/d81a6d8bba24b5569aa34818d126475ecc38c801) without issue.

**Review by jsf9k — APPROVED** [review]
Strong work!

**Comment by dav3r** [comment]
@mcdonnnj I believe this PR is just waiting on your re-review.

**Comment by jsf9k** [comment]
@mcdonnnj - Why is Lineage not generating a PR for this repo?  The `develop` branch is missing the latest commits from the skeleton.

**Comment by mcdonnnj** [comment]
@king-alexander Please either rebase on or merge `develop` now that #12 has been merged. This should put you in position to pass tests with this PR.
